### ISSUE TITLE
Using part of testdata as validation?

### ISSUE THREAD

**Issue by guthasaibharathchandra** [IBM/concept_transformer#3]
Hi, thank you very much for publishing your code, I am following up on your work and so I'm trying to reproduce the results from your paper for cub dataset. In this process, I found that the model uses a part of test-data as validation data and also since it is saving best model checkpoint monitoring the celoss on validation dataset, I'm wondering how we could do this, since we are leaking some info from test into our training pipeline.

https://github.com/IBM/concept_transformer/blob/edfe59b1fcb2d5e5a627a3f81e50f54345bba2ba/data/cub2011parts_datamodule.py#L47

I might be missing something, kindly let me know. Thanks!

**Comment by matrig** [comment]
Hi, thanks for your interest in our work!
Yes, the confusion stems from the fact that we didn't publish the loop for hyperparameters tuning in the repo. The code only runs the training on one fixed set of hyperparameters (i.e. that is not being optimized) and we therefore don't distinguish between validation and test set. I hope this clarifies the issue.

**Comment by guthasaibharathchandra** [comment]
Thanks for your quick response. But since we finally test the model on the entire test-data (which now also includes the validation data because the valid split is made from this test-data),  my concern was how fair can this evaluation on test-set be? (if we already used part of test-data i.e valid-data to checkpoint our best model) 
  https://github.com/IBM/concept_transformer/blob/edfe59b1fcb2d5e5a627a3f81e50f54345bba2ba/data/cub2011parts_datamodule.py#L51

**Comment by matrig** [comment]
Oh right. You shouldn't use the best checkpoint on the validation split (since it overlaps with the test split). Either you remove a subset from the training set and use that as validation set, or just pick the last checkpoint (instead of the best).

**Comment by guthasaibharathchandra** [comment]
Thanks for the quick clarification!

**Comment by guthasaibharathchandra** [comment]
Hi, Is it possible to provide the hyper-parameter settings that achieved the final results reported in the paper? I noticed that they were not mentioned in the paper either. Thanks!

**Comment by matrig** [comment]
I increased the default weight_decay to match what we used for the paper (just pushed it). Other than that, the code in the repo should reproduce the results in the paper. So, the last checkpoint at epoch 50 should match the performance we reported. Feel free to let me know if that is not the case.
